internal/query/org.go

```diff
@@ -0,0 +1,236 @@
+					&count,
+				)
+				if err != nil {
+					return nil, err
+				}
+				orgs = append(orgs, org)
+			}
+
+			if err := rows.Close(); err != nil {
+				return nil, errors.ThrowInternal(err, "QUERY-QMXJv", "unable to close rows")
+			}
+
+			return &Orgs{
+				Orgs:  orgs,
+				Count: count,
+			}, nil
+		}
+}
+
+func (q *Queries) prepareOrgUniqueQuery() (sq.SelectBuilder, func(*sql.Row) (bool, error)) {
+	return sq.Select("COUNT(*) = 0").
+			From(orgTable).PlaceholderFormat(sq.Dollar),
+		func(row *sql.Row) (isUnique bool, err error) {
+			err = row.Scan(&isUnique)
+			if err != nil {
+				return false, errors.ThrowInternal(err, "QUERY-pWS5H", "errors.internal")
+			}
+			return isUnique, err
+		}
+}
+
+func (q *Queries) OrgByID(ctx context.Context, id string) (*Org, error) {
+	stmt, scan := prepareOrgQuery()
+	query, args, err := stmt.Where(sq.Eq{
+		OrgColumnID.toColumnName(): id,
+	}).ToSql()
+	if err != nil {
+		return nil, errors.ThrowInternal(err, "QUERY-AWx52", "unable to create sql stmt")
+	}
+
+	row := q.client.QueryRowContext(ctx, query, args...)
+	return scan(row)
+}
+
+func (q *Queries) OrgByDomainGlobal(ctx context.Context, domain string) (*Org, error) {
+	stmt, scan := prepareOrgQuery()
+	query, args, err := stmt.Where(sq.Eq{
+		OrgColumnDomain.toColumnName(): domain,
+	}).ToSql()
+	if err != nil {
+		return nil, errors.ThrowInternal(err, "QUERY-TYUCE", "unable to create sql stmt")
+	}
+
+	row := q.client.QueryRowContext(ctx, query, args...)
+	return scan(row)
+}
+
+func (q *Queries) IsOrgUnique(ctx context.Context, name, domain string) (isUnique bool, err error) {
+	query, scan := q.prepareOrgUniqueQuery()
+	stmt, args, err := query.Where(sq.Eq{
+		OrgColumnDomain.toColumnName(): domain,
+	}).ToSql()
+	if err != nil {
+		return false, errors.ThrowInternal(err, "QUERY-TYUCE", "unable to create sql stmt")
+	}
+
+	row := q.client.QueryRowContext(ctx, stmt, args...)
+	return scan(row)
+}
+
+func (q *Queries) ExistsOrg(ctx context.Context, id string) (err error) {
+	_, err = q.OrgByID(ctx, id)
+	return err
+}
+
+func (q *Queries) SearchOrgs(ctx context.Context, queries *OrgSearchQueries) (orgs *Orgs, err error) {
+	query, scan := q.prepareOrgsQuery()
+	stmt, args, err := queries.toQuery(query).ToSql()
+	if err != nil {
+		return nil, errors.ThrowInvalidArgument(err, "QUERY-wQ3by", "Errors.orgs.invalid.request")
+	}
+
+	rows, err := q.client.QueryContext(ctx, stmt, args...)
+	if err != nil {
+		return nil, errors.ThrowInternal(err, "QUERY-M6mYN", "Errors.orgs.internal")
+	}
+	return scan(rows)
+}
+
+type Orgs struct {
+	Count uint64
+	Orgs  []*Org
+}
+
+type Org struct {
+	ID            string          `col:"id"`
+	CreationDate  time.Time       `col:"creation_date"`
+	ChangeDate    time.Time       `col:"change_date"`
+	ResourceOwner string          `col:"resource_owner"`
+	State         domain.OrgState `col:"org_state"`
+	Sequence      uint64          `col:"sequence"`
+
+	Name   string `col:"name"`
+	Domain string `col:"domain"`
```

Comment:
I would rename it to primary_domain so it is clear what it is.

Reply:
i renamed the column and removed the tags as they are useless